tests/programs/dlopen_demo.sh

```diff
@@ -6,17 +6,35 @@
 # Copyright The Mbed TLS Contributors
 # SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
 
-. "${0%/*}/../../scripts/demo_common.sh"
+DEMO_COMMON_NEED_QUERY_COMPILE_TIME_CONFIG=0
+
+SCRIPT_DIR=$(dirname "$0")
+. "${SCRIPT_DIR}/../../scripts/project_detection.sh"
+. "${SCRIPT_DIR}/../../scripts/demo_common.sh"
 
 msg "Test the dynamic loading of libmbed*"
 
-program="$programs_dir/test/dlopen"
-library_dir="$root_dir/library"
+# Once demo_common.sh is sourced we'll have the following variables set:
+# - $root_dir points to the root path of Mbed TLS or TF-PSA-Crypto;
+# - $programs_dir points to "$root_dir/programs" folder.
+if is_mbedtls_root $root_dir; then
+    msg "Running in Mbed TLS repo"
+    program="$programs_dir/test/dlopen"
+    library_dir="$root_dir/library"
+else
+    msg "Running in TF-PSA-Crypto repo"
+    build_dir="."
+    if [ -d "$root_dir/out_of_source_build" ]; then
```

Comment:
`out_of_source_build` is all.sh specific thus we cannot really do that here. I think I understand why you do that though and I would rather suggest to just build in tree in `component_test_tf_psa_crypto_shared`.